quil-rs/src/instruction/pragma.rs

```diff
@@ -77,3 +77,26 @@ impl Include {
         Self { filename }
     }
 }
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum ReservedPragma {
```

Comment:
The main alternative considered here was to convert `Pragma` into an `enum Pragma { Free(...), Reserved(enum ReservedPragma()) }`. This feels somewhat more semantically clear, but:

1. This alternative still requires the `enum ReservedPragma`.
2. We create an additional layer of enums, rather than hoisting it up to the pre-existing `enum Instruction` level. Note, there is precedence for this: `Instruction::Arithmetic` actually maps to several different Quil commands (eg `ADD`, `SUB`, etc).

Generally, the overall justification for the approach is: adding enum variants is an easier change to handle, rather than converting a `struct` to an `enum`.

I don't feel incredibly strong about this, however.

Reply:
I think I'd prefer

```
enum Pragma {
    Extern { … },
    Unknown { // Or `Custom`, or `Generic`, or …
        pub name: String,
        pub arguments: Vec<PragmaArgument>,
        pub data: Option<String>,
    },
}
```

as it captures pragmas fairly cleanly, but I think this may relate to the question of splitting the parsed and resolved ASTs